chai_lab/data/io/cif_utils.py

```diff
@@ -96,27 +97,30 @@ def get_chains_metadata(
         any_token_in_resi = residue_indices.new_zeros([max_res + 1]) - 99
 
         any_token_in_resi[residue_indices] = torch.arange(
-            len(residue_indices),
-            dtype=residue_indices.dtype,
-            device=residue_indices.device,
+            len(residue_indices), dtype=any_token_in_resi.dtype
         )
 
         assert any_token_in_resi.min() >= 0
 
         sequence = [chain_token_res_names[i] for i in any_token_in_resi]
 
-        chain_id_str = _get_chain_letter(asym_id)
+        entity_key = (entity_type, *sequence)
```

Comment:
I think this runs into the issue we saw previously where the "sequence" of a ligand is just `LIG` so all ligands would map to the same entity key of `(EntityType.LIGAND.value, "LIG")`.